bot/exts/moderation/infraction/infractions.py

```diff
@@ -664,6 +645,26 @@ async def cog_command_error(self, ctx: Context, error: Exception) -> None:
                 await ctx.send(str(error.errors[0]))
                 error.handled = True
 
+    @commands.Cog.listener()
+    async def on_member_join(self, member: Member) -> None:
+        """
+
+        Apply active timeout infractions for returning members.
+
+        This is needed for users who might have had their infraction edited in our database but not in Discord itself.
+        """
+        active_timeouts = await self.bot.api_client.get(
+            endpoint="bot/infractions",
+            params={"active": "true", "type": "timeout", "user__id": member.id}
+        )
+        if active_timeouts and not member.is_timed_out():
+            reason = f"Applying active timeout for returning member: {active_timeouts[0]['id']}"
+
+        async def action() -> None:
+            await member.edit(timed_out_until=arrow.get(active_timeouts[0]["expires_at"]).datetime, reason=reason)
```

Comment:
This line triggers an `IndexError` whenever a user without an active timeout infraction joins the server.

Reply:
Good find, both were fixed in the [force push](https://github.com/python-discord/bot/compare/67b8c8ab641529d90aa39756e69e980b41428387..36ee066ff883d56ef31d129a45d37e35a18849ab)